pkg/core/ratelimits/matcher.go

```diff
@@ -57,6 +57,10 @@ func buildRateLimitMap(
 	return result, nil
 }
 
+// We want the rateLimits slice to be split into separate policies by source
+// I.e. if a RateLimit has two matches it its source, it will be cloned into two
+// RateLimit resources, each of them having only a single Source Match.
+// We rely on this later to sort the rateLimits using ConnectionPolicyBySourceService.
```

Comment:
Maybe it's nitpicking but that's `what` not `why`. How about
```
We need to split policies with many sources into individual policies, because otherwise we would not be able to sort all selectors from all policies in the most specific to the most general order. Example
  sources:
    - match:
        kuma.io/service: 'web'
        version: '1'
    - match:
        kuma.io/service: 'web-api'
---
  sources:
    - match:
        kuma.io/service: 'web'
    - match:
        kuma.io/service: 'web-api'
        version: '1' 
```